src/App.jsx

```diff
@@ -14,33 +17,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function internalArrayOfObjectsWithId(arr) {
+  const internalArrayOfObjects = [];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  for (let i = 0; i < arr.length; i += 1) {
+    internalArrayOfObjects.push({
+      name: arr[i],
+      internalId: i,
+    });
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  return internalArrayOfObjects;
+}
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+const goodsListWithId = internalArrayOfObjectsWithId(goodsFromServer);
+const SORT_BY_LENGTH = 'length';
+const SORT_BY_ABC = 'abc';
+
+export const App = () => {
+  const [sortValue, setSortValue] = useState('');
+  const [reversed, setReversed] = useState(false);
+
+  let sortedGoods = goodsListWithId.toSorted((good1, good2) => {
```

Comment:
The `toSorted` method does not exist on JavaScript arrays. You should use the `sort` method instead, which sorts the array in place. However, note that `sort` mutates the original array, so you might want to create a copy of the array before sorting if immutability is needed.